libraries/aws/shadow/include/shadow.h

```diff
@@ -0,0 +1,455 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file shadow.h
+ * @brief user-facing Shadow functions, and parameter structs.
+ */
+
+#ifndef _SHADOW_H_
+#define _SHADOW_H_
+
+/* Standard includes. */
+#include <stdint.h>
+/* Include config file before other headers. */
+#include "shadow_config.h"
+
+/*--------------------------- Shadow types ---------------------------*/
+
+/**
+ * @brief Each of these values describe the type of a shadow message.
+ *        https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html
+ */
+typedef enum ShadowMessageType
+{
+    ShadowMessageTypeGetAccepted = 0,
+    ShadowMessageTypeGetRejected,
+    ShadowMessageTypeDeleteAccepted,
+    ShadowMessageTypeDeleteRejected,
+    ShadowMessageTypeUpdateAccepted,
+    ShadowMessageTypeUpdateRejected,
+    ShadowMessageTypeUpdateDocuments,
+    ShadowMessageTypeUpdateDelta,
+    ShadowMessageTypeMaxNum
+} ShadowMessageType_t;
+
+/**
+ * @brief Each of these values describe the type of a shadow topic string.
```

Comment:
Nit: grammar
```suggestion
 * @brief Each of these values describes the type of a shadow topic string.
```

Reply:
fixed it in commit [a870748](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1081/commits/a870748a7b768e378c62babc6699ecc58115c539)

resolving this comment

feel free to let me know if it needs further changes, thank you